Timer/ViewController.swift

```diff
@@ -9,10 +9,213 @@ import UIKit
 
 class ViewController: UIViewController {
 
+    private lazy var parentStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.spacing = Metric.parentStackViewSpacing
+
+        return stackView
+    }()
+
+    //     MARK: - Life cycle
+
     override func viewDidLoad() {
         super.viewDidLoad()
+        animationCircle()
+        setupHierarchy()
+        setupLayout()
+        setupView()
+
+        timerButton.addTarget(self, action: #selector(actionTimerButton), for: .touchUpInside)
+        updateTimerText()
+    }
+
+    @objc func actionTimerButton() {
+        if isStarted {
+            if isPaused {
+                timer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(updateTimer), userInfo: nil, repeats: true)
+                isPaused = false
+                updateTimerButton()
+                resumeAnimation()
+            } else {
+                timer.invalidate()
+                isPaused = true
+                updateTimerButton()
+                pauseAnimation()
+            }
+        } else {
+            timer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(updateTimer), userInfo: nil, repeats: true)
+            timerButton.setImage(pauseButtonImage, for: .normal)
+            isStarted = true
+            isPaused = false
+            updateTimerButton()
+            basicAnimation()
+        }
+    }
+
+    private func setupView() {
+        view.backgroundColor = .white
+    }
+
+    private func setupHierarchy() {
+        view.layer.addSublayer(shapeLayer)
+        view.addSubview(parentStackView)
+
+        parentStackView.addArrangedSubview(timerLabel)
+        parentStackView.addArrangedSubview(timerButton)
+    }
+
+    private func setupLayout() {
+        parentStackView.translatesAutoresizingMaskIntoConstraints = false
+        parentStackView.clipsToBounds = true
+
+        parentStackView.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        parentStackView.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
+    }
+
+//     MARK: - Flags
+
+    var isStarted = false
+    var isWorkLap = true
+    var isPaused = false
+
+//     MARK: - Private func
+
+    private lazy var timerLabel: UILabel = {
+        let label = UILabel()
+        label.font = .systemFont(ofSize: Metric.timerFont)
+        label.textAlignment = .center
+        label.textColor = .red
+
+        return label
+    }()
+
+    private lazy var timerButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.setImage(playButtonImage, for: .normal)
+        button.tintColor = .red
+
+        return button
+    }()
+
+    private func createSystemImage(name: String) -> UIImage {
+        guard let image = UIImage(systemName: name)
+            else {
+                return UIImage.remove }
```

Comment:
Оформление блока `guard` лучше делать вот так:
```suggestion
        guard let image = UIImage(systemName: name) else {
                return UIImage.remove
        }
```